client/clip_client/client.py

```diff
@@ -119,6 +121,11 @@ def encode(self, content, **kwargs):
                 **self._get_post_payload(content, kwargs),
                 on_done=partial(self._gather_result, results=results),
             )
+
+        for c in content:
+            if isinstance(c, Document) and c.tags.pop('__loaded_by_CAS__', False):
```

Comment:
```suggestion
            if hasattr(c, 'tags') and c.tags.pop('__loaded_by_CAS__', False):
```

Reply:
This is a broken change

Reply:
The benefits from change above: 
1. reduce the memory footprint;
2. keep the inputs not updated;

Reply:
The future better way is to support **in-place update** in `encode(docs)`